src/mlpack/core/cereal/array_wrapper.hpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file core/cereal/array_wrapper.hpp
+ * @author Omar Shrit
+ *
+ * Implementation of an array wrapper.
+ *
+ * This implementation allows to serilize array easily using cereal.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CEREAL_ARRAY_WRAPPER_HPP
+#define MLPACK_CORE_CEREAL_ARRAY_WRAPPER_HPP
+
+#include <cereal/archives/binary.hpp>
+#include <cereal/archives/portable_binary.hpp>
+#include <cereal/archives/xml.hpp>
+#include <cereal/archives/json.hpp>
+
+namespace cereal {
+
+template<class T>
+class ArrayWrapper
+{
+/* 
+ * This file add make_array functionality to cereal
+ */
+ public:
+  ArrayWrapper(T*& addr, std::size_t& size) :
+      arrayAddress(addr),
+      arraySize(size)
+  {}
+
+  /*
+  * Cereal does not require to split member, it can do that internally
+  * If this is the case we can not implement optimized version, since
+  * the only possible one is optimized.
+  * Some verification needed...
+  * default implementation
+  */
```

Comment:
I don't understand the comment; maybe it is inaccurate or needs to be clarified?

Reply:
This is a very old comment, it needs to be removed I will handle it